File: Easy/PHP/MinimumWaitingTime.php
<?php

namespace Easy\PHP\MinimumWaitingTime;

/**
 * Returns the minimum waiting time.
 *
 * O(nlog(n)) time | O(1) space.
 *
 * @param array $queries  The input array.
 *
 * @return integer minimum waiting time.
 */
function minimumWaitingTime($array)
{
    sort($array);
    $sum = 0;
    for ($i = 0; $i < count($array); $i++) {
        $sum += $array[$i] * ((count($array) - 1) - $i);
    }

    return $sum;
}

$queries = [17, 4, 3];
$result = minimumWaitingTime($queries);
//var_dump($result);
print_r($result);

File: Easy/JAVA/MinimumWaitingTime.java
package Easy.JAVA;

import java.util.Arrays;

public class MinimumWaitingTime {

    // O(nlog(n)) | O(1)
    private static int waitingTime(int[] queries) {
        Arrays.sort(queries);
        var sum = 0;
        for (int i = 0; i < queries.length; i++) {
            sum += queries[i] * ((queries.length - 1) - i);
        }
        return sum;
    }

    public static void main(String[] args) {

        var times = new int[] { 17, 4, 3 };
        var result = waitingTime(times);
        System.out.println(result);
    }
}
